test/unit/plugins/rest_spec.js

```diff
@@ -134,5 +139,237 @@ describe("The Rest plugin", function () {
+				var payload;
+				expect(result.statusCode,"status").to.equal(200);
+				payload = JSON.parse(result.payload);
+				expect(payload.instances.length).equal(0);
+			});
+		});
+
+		describe("with all invalid parameters for a GET Request", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?extra=test&random=thisiswrong")
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns a Bad Request", function () {
+				expect(result.statusCode,"status").to.equal(400);
+			});
+		});
+
+		describe("with a mix of valid and invalid parameters", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?extra=test&random=thisiswrong&type=" + VALID_TYPE)
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns a Bad Request", function () {
+				expect(result.statusCode,"status").to.equal(400);
+			});
+		});
+
+		describe("with a valid ID", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?id=" + VALID_INSTANCE_ID)
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an instance with requested id", function () {
+				var payload;
+				expect(result.statusCode,"status").to.equal(200);
+				payload = JSON.parse(result.payload);
+				expect(payload.instances.length).equal(1);
+				expect(payload.instances[ 0 ].id).to.equal(VALID_INSTANCE_ID);
+			});
+		});
+
+		describe("with a valid AMI", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?ami=" + VALID_AMI)
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an array of instances with requested ami", function () {
+				expect(result.statusCode,"status").to.equal(200);
+			});
+		});
+
+		describe("with state=pending", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?state=pending")
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an array of instances which are in a pending state", function () {
+				expect(result.statusCode,"status").to.equal(200);
+			});
+		});
+
+		describe("with invalid URI", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?uri=" + INVALID_QUERY)
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an empty array of intsances", function () {
+				var payload;
+				expect(result.statusCode,"status").to.equal(200);
+				payload = JSON.parse(result.payload);
+				expect(payload.instances.length).equal(0);
+			});
+		});
+	});
+	describe("when registered with stubbed instance", function () {
```

Comment:
This description may be a little too specific to your test implementation. Descriptions work better if they are more... generally descriptive, instead of prescribing a certain way of doing things. For instance, `"encountering an internal error"` might work here, since these test cases are for 500 error situations. You accomplish the tests by stubbing, but that's not necessarily relevant to the test results output.